### PR TITLE
Patch for python3 and archlinux support.

### PR DESCRIPTION
**Pull request by zupm**
Add python2/python3 compatibility and detect archlinux distribution.

### CODE REVIEW

File: docker_mirror.py [hunk 1]
```diff
@@ -1,6 +1,8 @@
 #!/usr/bin/python
 # coding=utf8
 
+from __future__ import print_function
+from __future__ import unicode_literals
```
Comment (silenceshell) [inline docker_mirror.py:5]: Is this import necessary?

File: docker_mirror.py [hunk 2]
```diff
@@ -108,7 +113,7 @@ def set_docker_config_ce(mirror):
     config_dict={}
     if os.path.exists(docker_config) != True:
         # if file not exist, create it first.
-        os.mknod(docker_config, 0644)
+        os.mknod(docker_config, 0o644)
```
Comment (silenceshell) [inline docker_mirror.py:116]: What does the 'o' mean?

**Review by silenceshell — COMMENTED** [review]
hi @ZhangHuaiFan
Thanks for your PR. I have some questions, hopes you will answer them.

File: docker_mirror.py (hunk 2 above)
Comment (zupm) [reply docker_mirror.py:116]: For Python 2.6, the changed and new token definitions will be:

integer        ::=     decimalinteger | octinteger | hexinteger |
                       bininteger | oldoctinteger

**octinteger     ::=     "0" ("o" | "O") octdigit+**

File: docker_mirror.py (hunk 1 above)
Comment (zupm) [reply docker_mirror.py:5]: A future statement is a directive to the compiler that a particular module should be compiled using syntax or semantics that will be available in a specified future release of Python. The future statement is intended to ease migration to future versions of Python that introduce incompatible changes to the language.
[refer to this](https://docs.python.org/2/reference/simple_stmts.html#future)

**Pull request merged**